### PR TITLE
Feat/4-pipeline

### LINKED ISSUE #4: [FEAT] Build Training Pipeline
## Background
- Establish a common framework for recommender system pipelines to enhance efficiency and modularity before developing specialized models.

## To do
- [x] update training pipeline draft
- [x] train_config.yaml & train.py @twndus 
- [x] BaseTrainer.py & metric.py @GangBean

### PR DESCRIPTION
**Pull request by twndus**
## Overview
- Completed the construction of the training pipeline. After integrating this pipeline, initial models and modules that inherit from this pipeline should be implemented.

## Change Log
-

## To Reviewer
-

## Issue Tags
- Closed: #4 
- See also: #

### CODE REVIEW

File: models/base_model.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import torch
+from torch.nn import Module
+
+from abc import abstractmethod
+
+class BaseModel(Module):
```
Comment (GangBean) [inline models/base_model.py]: need to be changed into abstract class

**Review by GangBean — APPROVED** [review]
LGTM 👍

**Review by GangBean — APPROVED** [review]
